go/vt/vtgate/engine/vindex_func.go

```diff
@@ -125,16 +125,21 @@ func (vf *VindexFunc) mapVindex(ctx context.Context, vcursor VCursor, bindVars m
 	result := &sqltypes.Result{
 		Fields: vf.Fields,
 	}
-	for _, value := range values {
+	destinations, err := vf.Vindex.Map(ctx, vcursor, values)
+	if err != nil {
+		return nil, err
+	}
+	if len(destinations) != len(values) {
+		// should never happen
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "unique Vindex.Map() length mismatch: input values count is %d, output destinations count is %d",
```

Comment:
nit: remove `unique` from the error message.

Reply:
Yeah, I noticed that too. Thanks! Fixed in: https://github.com/vitessio/vitess/pull/11531/commits/0ce233cf2d80ed836f22f15ee66f8490111f82f6